shuup/admin/modules/services/object_selector.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shuup Commerce Inc. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Iterable, Tuple
+
+from shuup.admin.utils.permissions import has_permission
+from shuup.admin.views.select import BaseAdminObjectSelector
+from shuup.core.models import Carrier, PaymentMethod, ShippingMethod
+
+
+class CarrierAdminObjectSelector(BaseAdminObjectSelector):
+    ordering = 13
+
+    @classmethod
+    def handles_selector(cls, selector):
+        return selector == "shuup.carrier"
+
+    def has_permission(self, user):
+        return has_permission(user, "carrier.object_selector")
+
+    def get_objects(self, search_term, *args, **kwargs) -> Iterable[Tuple[int, str]]:
+        """
+        Returns an iterable of tuples of (id, text)
+        """
+        objects = list(
```

Comment:
Any business logic to add here?

Reply:
filter by the current shop. if there is a supplier, use that on the filter as well.